File: Gateway/src/main/java/com/quizGpt/Gateway/Quiz/Dto/UserAnswerDto.java
package com.quizGpt.Gateway.Quiz.Dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

@Data
@NoArgsConstructor
@AllArgsConstructor
@ToString
public class UserAnswerDto {
    public String userAnswerId; // uuid
    public int questionId;
    public String text;
}

File: resource-server/src/dtos/userAnswerDto.ts
import { QuizAttemptDto } from "./quizAttemptDto.js";

export class UserAnswerDto {
  userAnswerId: number;
  quizAttempt: QuizAttemptDto;
  text: String;

  constructor(userAnswerId: number, quizAttempt: QuizAttemptDto, text: String) {
    this.userAnswerId = userAnswerId;
    this.quizAttempt = quizAttempt;
    this.text = text;
  }
}
